tests/manage/z_cluster/test_log_rotate.py

```diff
@@ -0,0 +1,186 @@
+import logging
+import time
+import pytest
+import re
+
+from ocs_ci.ocs import defaults
+from ocs_ci.ocs.resources.storage_cluster import verify_storage_cluster
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier2,
+    bugzilla,
+    skipif_ocs_version,
+    skipif_external_mode,
+    ignore_leftovers,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@ignore_leftovers
+@bugzilla("2116416")
+@skipif_external_mode
+@skipif_ocs_version("<4.10")
+@pytest.mark.polarion_id("OCS-4684")
+class TestLogsRotate(ManageTest):
+    """
+    Test Logs Rotate
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            log.info("Delete logCollector from storage cluster yaml file")
+            storagecluster_obj = OCP(
+                resource_name=constants.DEFAULT_CLUSTERNAME,
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                kind=constants.STORAGECLUSTER,
+            )
+            params = '[{"op": "remove", "path": "/spec/logCollector"}]'
+            storagecluster_obj.patch(params=params, format_type="json")
+            log.info(
+                "It takes time for storagecluster to update after the edit command"
+            )
+            time.sleep(30)
+            log.info("Verify storagecluster on Ready state")
+            verify_storage_cluster()
+
+        request.addfinalizer(finalizer)
+
+    def get_pod_obj_based_on_id(self, pod_type):
+        """
+        Get Pod Obj
+
+        Args:
+            pod_type (str): The type of pod [osd/mon/mgr/rgw/mds]
+
+        Returns:
+            POD Obj:
+        """
```

Comment:
dont understand why you're doing this.
you can simply Pod(**get_mds_pods[0]) to get the pod object?

Reply:
@mashetty330 After edit storagecluster all the pods are respined so the names change. The goal is to  work on the same pods.  
get_mds_pods[0] -> we don't know if the list is sorted...